onnxruntime/core/graph/graph.cc

```diff
@@ -48,22 +48,6 @@ namespace onnxruntime {
     GraphProtoSyncNeeded(sync_needed);               \
   } while (0)
 
-static bool UsingLatestOnnxOpset(const DomainToVersionMap& opset_versions) {
-  bool is_latest_opset = false;
```

Comment:
Why is the behavior of always being strict for the latest opset being changed? 

It was added to help catch issues with onnx shape inferencing asap, so if we turn this off it would mean a unit test author adding a new test for an updated op would need to know to manually set this to strict to validate the shape inferencing. 

Assuming we re-instate the check, can we make the user provided 'strict' value std::optional so we only apply it if explicitly specified?